kiwi/boot/image/dracut.py

```diff
@@ -74,7 +90,7 @@ def include_module(self, module, install_media=False):
         elif module not in self.modules:
             self.modules.append(module)
 
-    def omit_module(self, module, install_media=False):
+    def omit_module(self, module: str, install_media: bool = False) -> None:
```

Comment:
I wonder why we want to classify the return of a method that returns nothing with an explicit "None" type.
I see where you coming from and I was also into C and C++ for some time. Also in that language I found it cumbersome to explicitly state "void" when it's obvious that no return type definitions means we don't return
anything. Does this provide as any benefit ? meaning if an api method is used that is not expected to be called with an evaluated return value, does python complain before ? e.g

```python
a = omit_module(...)
```

would python complain before running the code if the method definition says "-> None" ? If not I don't see any value ?

Reply:
Python itself won't complain, because python doesn't care about type hints. What cares about them is additional tooling (like mypy).

The idea of `func() -> None:` is to be the equivalent of `void func();` in C. I.e. this function doesn't return anything and saving it's return value is thus most likely a bug. If you only return `None` in some cases, then the return type should be `Option[T]` (and then mypy will also complain if you just assume that this value is always not `None`).

Reply:
If we agree to base the type hints on py >= 3.6 as discussed a bit on the rio chat, would this still apply ?

Reply:
Yes, a function that returns nothing (= equivalent to `void func();` in C) should be marked as `func() -> None:` for Python type hints.